addons/product/tests/test_product_attribute_value_config.py

```diff
@@ -381,12 +381,9 @@ def test_get_first_possible_combination(self):
         self._add_exclude(computer_ram_32, computer_ssd_256)
         self.assertEqual(self.computer._get_first_possible_combination(), computer_ssd_512 + computer_ram_32 + computer_hdd_4)
 
-        # No possible combination (test helper and iterator)
-        self._add_exclude(computer_ram_32, computer_hdd_4)
-        self.assertEqual(self.computer._get_first_possible_combination(), self.env['product.template.attribute.value'])
-        gen = self.computer._get_possible_combinations()
-        with self.assertRaises(StopIteration):
-            next(gen)
+        # Not possible to add an exclusion when only one variant is left -> it deletes the product template associated to it
+        with self.assertRaises(UserError), self.cr.savepoint():
+            self._add_exclude(computer_ram_32, computer_hdd_4)
```

Comment:
Ok for this assert, but ideally it would be nice to keep an assert covering `_get_first_possible_combination` too. (should be doable with dynamic attributes as it doesn't delete the template)

Reply:
Would two next(gen) in a row be ok? as in new commit

Reply:
It's not testing `_get_first_possible_combination` so I would say no.

Reply:
And since it is tested all the way above such as in 
https://github.com/odoo/odoo/blob/2921be4b9199402f459ad32cb6028c7cba8d77d7/addons/product/tests/test_product_attribute_value_config.py#L382

Isn't it enough? If not, I'm not sure to understand why

Reply:
Ok, thanks! I'll change that

Reply:
I cannot change the attribute type:
`You cannot change the Variants Creation Mode of the attribute RAM because it is used on the following products:\nSuper Computer'`

So, I deleted all existing exclusions for `computer_ram_32` and created one line with all exclusions (this flow does not trigger the deletion of the template). Now, the test works.

Reply:
Let me just change so we test forst_possible